sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -958,6 +960,7 @@ export class ManagementClient extends LinkEntity {
       const request: AmqpMessage = {
         body: messageBody,
         reply_to: this.replyTo,
+        message_id: generate_uuid(),
```

Comment:
The construction of this message feels like a good candidate to be in a function. The only varying parts are the operation from what I can see in this brief diff.

Reply:
Also, I'd be worried that we could easily forget this in some other spot.

Reply:
> Also, I'd be worried that we could easily forget this in some other spot.

Which is exactly why we are adding the backup message_id at `core-amqp` level. :P

Reply:
This code should be defensive though and enforce it's own constraints, right?

Reply:
The change to core-amqp takes care of adding a message id if one forgets it in this layer. We can take the approach of not setting the message id for any of the operations here and rely solely on what core-amqp does. But then, we should update the min dep for core-amqp

Reply:
That makes sense too. Whomever is responsible for it should validate it's happening everywhere consistently :)

Reply:
Follow up question - does this change make it possible for us to return the generated ID to the customer? That's the only reason I could see potentially favoring one vs the other.

Reply:
We can tack it onto the response returned at core-amqp, pass it to the SDK and to the user.
This need not be the deciding factor on who's responsible.

Reply:
Cool, just wondering. We don't have a reason to do it yet so that was just curiosity on my part :)

Reply:
No, this does not change anything regarding service generated message id to customers because the "message" we are talking about here is the one we "make up" in order to send as part of the request/response pattern. The body of this made up message is what the user would care about which we dont touch

Reply:
Ah, I see. This is just the envelope.

Reply:
But the user was asking about the request-id the SDK sets, isn't it?

Reply:
It was, but for `sender.send()`. So not the management link.

Reply:
We can do 1 of two things:
- If we update the min version of core-amqp to `1.1.3`, then we can skip setting the messageId altogether in this file
- Else, we update the `_makeManagementRequest` method in this file to be the only place from which the message id is set

Reply:
Going with Else.

Reply:
Updated!